File: Examples/test-suite/javascript/static_member_overload_runme.js
const static_member_overload = require('static_member_overload');

const foo = new static_member_overload.Foo;

if (foo.sum(1, 2) !== 3) throw new Error('instance member failed');
if (static_member_overload.Foo.sum(1, 2, 3) !== 6) throw new Error('static member failed');

try {
  foo.sum(1, 2, 3);
  throw new Error('instance member should have failed');
} catch {}

try {
  static_member_overload.Foo.sum(1, 2);
  throw new Error('instance member should have failed');
} catch {}

File: Examples/test-suite/python/static_member_overload_runme.py
from static_member_overload import *

foo = Foo()

# TODO
# The current dispatcher is not able to handle both
# instance and static methods
if foo.sum(foo, 1, 2) != 3:
  raise RuntimeError('instance member failed')

if Foo.sum(1, 2, 3) != 6:
  raise RuntimeError('static member failed')

try:
  foo.sum(1, 2, 3)
  raise RuntimeError('instance member should have failed')
except:
  pass

try:
  Foo.sum(1, 2)
  raise RuntimeError('static member should have failed')
except:
  pass
